client/me/form-base/with-form-base.jsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:me:form-base' );
+
+/**
+ * Internal dependencies
+ */
+import notices from 'calypso/notices';
+import user from 'calypso/lib/user';
+
+const withFormBase = ( WrappedComponent ) => {
+	return class extends React.Component {
+		static displayName = `withFormBase(${ WrappedComponent.displayName || WrappedComponent.name })`;
+
+		state = {
+			redirect: false,
+			submittingForm: false,
+			changingUsername: false,
+			usernameAction: 'new',
+			showNotice: false,
+		};
+
+		componentDidMount() {
+			this.props.userSettings.getSettings();
```

Comment:
Not related to the HOC migration. The prop is passed from the route controller. It will be gone as soon as we migrate to Redux. I have a PR open where I remove some of the `userSettings` props where it's not necessary: #48443 – maybe that shades some light?

Reply:
I understand where it comes from, but it's undesired to access a prop that will not be there if it's not specifically passed to the component that the HoC is wrapping. In other words, relying on the fact it will always be there isn't reliable. Does that make sense?

Reply:
You're absolutely right. In this case, let's import it within this HOC to make sure it's there as long as we use it this way.

Reply:
fixed via c18ffaa